### ISSUE TITLE
Rere...reresending emails, but giving feedback that sending failed

### ISSUE THREAD

**Issue by cbz20** [mailpile/Mailpile#2014]
Earlier tonight, I sent an email with a ~3MB attachment, but noticed repeated messages 

"Could not send mail to \<one of the recipients' email address\>".

Later, I got a message from one of the recipients telling me (half annoyed, half amused) that his inbox had been flooded by about 30-40 identical copies of the original email from me sent at an interval of 1-5 minutes. 

I have now done some more tests (with another recipient, of course, namely my second email address). I have successfully reproduced the issue with the same attachment (some zip-file) and also with a different one of about the same size (some xcf-file).

Removing the outbox label from the message stops the resending of messages, returning a message "Sending cancelled". 

I am testing rc2.

**Comment by BjarniRunar** [comment]
Thanks for reporting this! Could you clarify, the person who received multiple copies - is that the same e-mail as Mailpile thought it couldn't deliver to?

**Comment by cbz20** [comment]
You're welcome! The answer is yes, that was the same person/email address. I suspect however that the other recipients also got multiple copies, but they have not replied yet. All I can say at the moment is that those multiple copies are on my outgoing mail server.

When I reproduced the issue with my own alternative email address, I only looked into the sending mail server where multiple messages kept appearing. However, on the recipient server, I only found one copy. I find this rather puzzling at the moment.

**Comment by BjarniRunar** [comment]
I need to review the code; my guess is either the SMTP server is broken here and is accepting the message for delivery without receiving the end-of-message marker, or Mailpile is timing out in spite of having already sent the end-of-message marker.

Of the two, I think a bug in Mailpile is far, far more likely.

**Comment by cbz20** [comment]
Thanks for the update. I just did the same experiment with a different sending server. The result is the same.

**Comment by BjarniRunar** [comment]
I've pushed up a change to the SMTP timeout logic which may help prevent this. Could you test and let me know how it goes?

**Comment by cbz20** [comment]
I am closing this because I cannot reproduce the bug any more: neither in the rc2 version that I use daily (and that I used when I first encountered the problem), nor in the latest nightly build that I just set up on a testing system. This is not particularly satisfying, but if the problem comes up again, one can still reopen the issue. 

Anyway, thanks a lot for your help, and sorry for the slow response!
